classes/UpgradeContainer.php

```diff
@@ -265,7 +265,7 @@ public function getProperty(string $property): string
             case self::LOGS_PATH:
                 return $this->autoupgradeWorkDir . DIRECTORY_SEPARATOR . 'logs';
             case self::ARCHIVE_FILENAME:
-                return $this->getUpgradeConfiguration()->getArchiveZip();
+                return $this->getUpgradeConfiguration()->getArchiveZip() ? $this->getUpgradeConfiguration()->getArchiveZip() : UpgradeConfiguration::DEFAULT_FILENAME;
```

Comment:
Can we find another place for setting the default value? I'm not convinced the container is the best place for that.

We can find the use of default value in the `Upgrader` class, it may be enough?

Reply:
This will cause a problem on `case self::ARCHIVE_FILEPATH`

Reply:
What will cause the issue? the current state or the requested change? It may be possible to handle this error even by moving the default value elsewhere.

Reply:
the requested change will pose a problem. You ask if the default value in upgrader can be enough, here we are going to have a problem on the `case self::ARCHIVE_FILEPATH` which itself refers to the `case self::ARCHIVE_FILENAME`.

Reply:
> It may be possible to handle this error even by moving the default value elsewhere.

We could put this elsewhere, but what's the point here?

Reply:
This is not the responsibility of the container to have the business logic, its purpose is to provide shortcuts to other parts of the codebase.

Reply:
I think that the naming of this "default" value is wrong, it is the name of the online archive. We are mixing the two concepts which are not appropriate here, we should review this whole part.

Reply:
> This is not the responsibility of the container to have the business logic, its purpose is to provide shortcuts to other parts of the codebase. 

However, there is business logic in several places in this class

> It looks confusing to have different methods with the same same

here I don't see 2 methods with the same name

Reply:
Yes, that's the name of the online archive. We can change the name of the constant if it improves understanding.

Reply:
If the location is a problem, then we would need an "intelligent" getter on the UpgradeConfiguration side, which would have this logic, and which would be called here. What do you think?

Reply:
It will be necessary to keep getArchiveZip() unchanged for the needs of the UI

Reply:
> However, there is business logic in several places in this class

My bad if this was introduced by me or if I let it go through recently. But as we tend to make this module better, let's avoid making the same mistakes.

> If the location is a problem, then we would need an "intelligent" getter on the UpgradeConfiguration side, which would have this logic, and which would be called here. What do you think?

This is indeed what I suggested. `UpgradeConfiguration` is a candidate, `Upgrader` could fit this need as well.
And yeah there is a scenario where if the merchants configures the archive channel, then goes back to the online channel: the value of this property would be wrong because it still returns the value of the archive zip file for the archive channel, even though we went back to the online option.
This code would likely changed to fit the proper name based on the selected channel, and this is at that moment you see this logic growing in complexity (even though it remains quite simple compared to the rest of the module). That's why it should be moved elsewhere.


About
>here I don't see 2 methods with the same name

`UpgradeContainer::getProperty(self::ARCHIVE_FILENAME)` and `UpgradeConfiguration::getArchiveZip()` have almost the same meaning, but their behavior are different. One returns the actual source file to use during the update process, the other one returns the file name configured for a specific channel. It seemed to be clear for someone who recently worked on this part, but from my point of view it was confusing before the difference was explained to me.

Reply:
> And yeah there is a scenario where if the merchants configures the archive channel, then goes back to the online channel: the value of this property would be wrong because it still returns the value of the archive zip file for the archive channel, even though we went back to the online option.

just condition this according to the channel,  and it will work as expected.

> UpgradeContainer::getProperty(self::ARCHIVE_FILENAME) and UpgradeConfiguration::getArchiveZip() have almost the same meaning, but their behavior are different. One returns the actual source file to use during the update process, the other one returns the file name configured for a specific channel. It seemed to be clear for someone who recently worked on this part, but from my point of view it was confusing before the difference was explained to me.

There will always be the `UpgradeConfiguration::getArchiveZip()` function, which will return the actual configured value, and the `UpgradeContainer::getProperty(self::ARCHIVE_FILENAME)` function, which will return the name of the archive depending on the channel. Adding a function in UpgradeConfiguration will not necessarily help in understanding for me. `getArchiveZip` should probably be renamed to `getLocalArchiveZip`